### PR TITLE
Added _DEBUG define

### PR DESCRIPTION
**Pull request by APerricone**
Added _DEBUG define to check if the current line is compiling with debug symbols

### CODE REVIEW

**Comment by alcz** [comment]
Good idea it seems. Sometimes in the past I wanted a thing like this, mostly in ad hoc/quick projects.

In bigger projects people probably already have some sort of solution, that defines something and /b compiler flag together.

I have doubt about the chosen ```_DEBUG``` may already clash with something user-defined. Likewise ```__DEBUG__``` and similar. Turns out hard to decide.

**Comment by APerricone** [comment]
I choosen _DEBUG for C++ compatibility, I think it is something that should was added decads ago.
Unlike user-definde it is dynamic in case of pragma, see sample code: https://github.com/APerricone/core/blob/_debug-predefine_without_SSL/tests/pp_DEBUG.prg

**Comment by alcz** [comment]
Yes, ```#pragma -b -``` will undefine ```_DEBUG``` even if someone has decided to define it "project-wide". That may be unexpected situation for somebody. C++ spelling is not that important for .prg code I think.

File: src/pp/ppcore.c [hunk 1]
```diff
@@ -5747,6 +5759,10 @@ void hb_pp_initDynDefines( PHB_PP_STATE pState, HB_BOOL fArchDefs )
 #ifdef HB_START_PROCEDURE
    hb_pp_addDefine( pState, "__HB_MAIN__", HB_START_PROCEDURE );
 #endif
+
+   if(fDebug)
```
Comment (alcz) [inline src/pp/ppcore.c]: Please note, that a valid pull request should also have usual code style/formatting.
```c
if( fDebug )
```
Here it should be with spaces. Another example few lines back.

**Comment by alcz** [comment]
For the debug #define name my proposals are (IMHO they fit more in the current set):
```
__HB_DEBUG__
__DEBUGINFO__
```
Lastly i could also find ```__DEBUG__``` acceptable, but would like to hear comments from other devs in this case.

**Comment by mlacecilia** [comment]
Hi,
IMHO __HB_DEBUG__ seems the best candidate, giving no chances to be
confused with C level debugging.
Best regards.
--
Maurizio

Il gio 19 mar 2020, 14:29 alcz <notifications@github.com> ha scritto:

**Comment by druzus** [comment]
Hi Antonio,
What is wrong with "#IF __paragma( DEBUGINFO )" or "#if __pragma( b )"?
IMHO it's redundant and user can set such definition yourself. See 
example below,

best regards
Przemek

proc main()
#if __pragma( DEBUGINFO )
    ? "compiled with debug -b option"
    #define _DEBUG
#else
    ? "compiled without debugher PCODE"
#endif
?
return


W dniu 19.03.2020 o 17:04, Maurizio la Cecilia pisze:

**Comment by APerricone** [comment]
Wow, many thanks Druzus, I did not know of the exists of this possibility... so my pull has no reason to be merged.

**Comment by alcz** [comment]
Thanks for reminder, its great! Seems like it was totally forgotten, with no example of __pragma( DEBUGINFO ) in current codebase, while ChangeLog.txt in fact informs about #if __pragma(...) generally.

**Pull request closed without merging**